update (GOPATH) paths to use github.com/moby/moby/v2

thaJeztah opened the pull request:
**- Human readable description for the release notes**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section.

NOTE: Only fill this section if changes introduced in this PR are user-facing.
The PR must have a relevant impact/ label.
-->
```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
update (GOPATH) paths to use github.com/moby/moby/v2

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

crazy-max commented on hack/make/.go-autogen:
```diff
@@ -1,12 +1,12 @@
--X \"github.com/docker/docker/dockerversion.ProductName=${PRODUCT}\" \
--X \"github.com/docker/docker/dockerversion.DefaultProductLicense=${DEFAULT_PRODUCT_LICENSE}\" "
+-X \"github.com/moby/moby/v2/dockerversion.Version=${VERSION}\" \
+-X \"github.com/moby/moby/v2/dockerversion.GitCommit=${GITCOMMIT}\" \
```
should be module path iirc

vvoland reviewed (commented):
I think we should use `github.com/moby/moby` as path (without the /v2 suffix).

See: https://github.com/moby/moby/pull/50622#discussion_r2251839664

dmcgowan commented:
> Yeah, not sure if we depend on anything still using GOPATH, but the correct path would be with v2 ?

Should just keep it at the `moby/moby` path, anything that does not handle the path would need to be handled specially, but changing the path will just break api and client for that. On containerd, it was only the proto generator that didn't consider Go modules, but we ended up just moving all the protos into the api anyway. Without that, just needed a step to put them in the right place after the proto runs.

thaJeztah committed:
docs/contributing: update paths to /usr/src/moby

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
use /usr/src/moby instead, and no GOPATH

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Looks like checkout-action doesn't like it on Windows;

```
Run actions/checkout@v6
  with:
    path: /usr/src/moby
    repository: moby/moby
    token: ***
    ssh-strict: true
    ssh-user: git
    persist-credentials: true
    clean: true
    sparse-checkout-cone-mode: true
    fetch-depth: 1
    fetch-tags: false
    show-progress: true
    lfs: false
    submodules: false
    set-safe-directory: true
  env:
    GO_VERSION: 1.25.7
    GOTESTLIST_VERSION: v0.3.1
    TESTSTAT_VERSION: v0.1.25
    WINDOWS_BASE_IMAGE: mcr.microsoft.com/windows/servercore
    WINDOWS_BASE_TAG_2022: ltsc2022
    WINDOWS_BASE_TAG_2025: ltsc2025
    TEST_IMAGE_NAME: moby:test
    TEST_CTN_NAME: moby
    DOCKER_BUILDKIT: 0
    ITG_CLI_MATRIX_SIZE: 6
    GOPATH: D:\a\moby\moby\go
    GOBIN: D:\a\moby\moby\go\bin
    BIN_OUT: D:\a\moby\moby\out
Error: Repository path 'D:\usr\src\moby' is not under 'D:\a\moby\moby'
```

thaJeztah committed:
gha: remove custom paths and working-directory

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

tianon commented:
Test failures are definitely real:

```
--- FAIL: TestPluginInstall (0.68s)
    --- FAIL: TestPluginInstall/no_auth (0.01s)
    --- FAIL: TestPluginInstall/with_digest (0.01s)
    --- FAIL: TestPluginInstall/with_htpasswd (0.01s)
    --- FAIL: TestPluginInstall/with_insecure (0.65s)
=== RUN   TestPluginsWithRuntimes
    plugin_test.go:280: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
--- FAIL: TestPluginsWithRuntimes (0.69s)
=== RUN   TestPluginBackCompatMediaTypes
    plugin_test.go:345: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
--- FAIL: TestPluginBackCompatMediaTypes (0.11s)
FAIL
coverage: [no statements]

=== Failed
=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall/no_auth (0.01s)
    plugin_test.go:114: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
    --- FAIL: TestPluginInstall/no_auth (0.01s)

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall/with_digest (0.01s)
    plugin_test.go:136: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
    --- FAIL: TestPluginInstall/with_digest (0.01s)

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall/with_htpasswd (0.01s)
    plugin_test.go:189: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
    --- FAIL: TestPluginInstall/with_htpasswd (0.01s)

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall/with_insecure (0.65s)
    plugin_test.go:245: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
    --- FAIL: TestPluginInstall/with_insecure (0.65s)

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall (0.68s)

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginsWithRuntimes (0.69s)
    plugin_test.go:280: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1

=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginBackCompatMediaTypes (0.11s)
    plugin_test.go:345: assertion failed: error is not nil: error building basic plugin bin: import "/usr/src/moby/internal/testutil/fixtures/plugin/basic": cannot import absolute path
        : exit status 1
```

tianon commented on internal/testutil/fixtures/plugin/plugin.go:
```diff
@@ -225,8 +225,8 @@ func ensureBasicPluginBin() (string, error) {
-	installPath := filepath.Join(os.Getenv("GOPATH"), "bin", name)
-	sourcePath := filepath.Join("github.com", "docker", "docker", "internal", "testutil", "fixtures", "plugin", "basic")
+	installPath := filepath.Join(os.Getenv("GOBIN"), name)
+	sourcePath := filepath.Join("/", "usr", "src", "moby", "internal", "testutil", "fixtures", "plugin", "basic")
```
This is both hard-coded to assumptions elsewhere (which makes it harder to run the tests outside our "blessed" environment -- not sure if anyone does that anymore) _and_ the exact absolute path the tests are complaining about (`go build` and friends don't accept absolute paths -- they accept module paths, and those are allowed to be relative or absolute, but absolute module paths look like what this code was using before, ie `foo/bar/baz`, not `/foo/bar/baz`).  In *theory* this should work fine with `github.com/moby/moby/v2/internal/testutil/fixtures/plugin/basic`, but I don't know for sure.

thaJeztah replied:
Ah, thanks! Yeah, I need to refresh my memory on the changes I made in this PR; I just did a rebase to see what state it was in, and I think some commits were still "lets' try if this works".

Might spend some time on it soon to see if we can simplify things (less assumptions about paths where go modules takes care of it).

thaJeztah commented:
Ah, derp; it worked locally, but I guess CI runs slightly different 😔😆

```
=== FAIL: amd64.usr.src.moby.integration.plugin.common TestPluginInstall/with_digest (0.05s)
    plugin_test.go:135: assertion failed: error is not nil: error building basic plugin bin: error obtaining VCS status: exit status 128
        	Use -buildvcs=false to disable VCS stamping.
        : exit status 1
    --- FAIL: TestPluginInstall/with_digest (0.05s)
```

thaJeztah committed:
Dockerfile.windows: remove redundant comment

This comment wasn't adding much, and it being a Dockerfile comment
(not a PowerShell comment) confused my IDE, flagging it as syntax
error.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
internal/testutil/fixtures/plugin: buildBasicPluginBin: don't use GOBIN

This function was assuming GOBIN would be set, but in many cases, it's not.
As a result, every call would build a new binary in the current working-
directory (and either build a new binary for each (sub)test, or write
over the binary when run in parallel.

Change it to use a tmpDir and a sync.OnceValues; this should prevent
unneeded rebuilds.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented on internal/testutil/fixtures/plugin/plugin.go:
```diff
@@ -214,21 +215,21 @@ func makePluginBundle(inPath string, opts ...CreateOpt) (io.ReadCloser, error) {
-	cmd := exec.Command(goBin, "build", "-o", installPath, sourcePath)
-	cmd.Env = append(os.Environ(), "CGO_ENABLED=0", "GO111MODULE=off")
+
+	tmpDir := filepath.Join(os.TempDir(), "moby-plugin-fixtures")
+	if err := os.MkdirAll(tmpDir, 0o755); err != nil {
+		return "", errors.Wrap(err, "error creating plugin fixtures dir")
+	}
+	installPath := filepath.Join(tmpDir, "docker-basic-plugin")
+	cmd := exec.Command(goBin, "build", "-buildvcs=false", "-o", installPath, "github.com/moby/moby/v2/internal/testutil/fixtures/plugin/basic")
+	cmd.Env = append(os.Environ(), "CGO_ENABLED=0", "GO111MODULE=auto")
```
That was somewhat intentional so that we replace the old one when building and not accumulate or each run. It's closer to the old (`GOBIN` based) location.

thaJeztah commented on docs/contributing/set-up-dev-env.md:
````diff
@@ -129,21 +129,21 @@ can take over 15 minutes to complete.
 
    ```none
    Successfully built 3d872560918e
-   docker run --rm -i --privileged -e BUILDFLAGS -e KEEPBUNDLE -e DOCKER_DEBUG -e DOCKER_EXPERIMENTAL -e DOCKER_GITCOMMIT -e DOCKER_GRAPHDRIVER=vfs -e DOCKER_REMAP_ROOT -e DOCKER_STORAGE_OPTS -e DOCKER_USERLANDPROXY -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "home/ubuntu/repos/docker/bundles:/go/src/github.com/docker/docker/bundles" -t "docker-dev" bash
+   docker run --rm -i --privileged -e BUILDFLAGS -e KEEPBUNDLE -e DOCKER_DEBUG ...
    #
````
I'm gonna stop fixing this one CoPilot. Once you get into your nit-pick stage you're never satisfied.
